core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -168,12 +172,27 @@ private void checkDiskSpace(DockerClient dockerClient, String id) {
         );
     }
 
-    private void checkMountableFile(DockerClient dockerClient, String id) {
-        try (InputStream stream = dockerClient.copyArchiveFromContainerCmd(id, "/dummy").exec()) {
-            stream.read();
-            VisibleAssertions.pass("File should be mountable");
+    private boolean checkMountableFile() {
+        DockerClient dockerClient = client();
+
+        MountableFile mountableFile = MountableFile.forClasspathResource(ResourceReaper.class.getName().replace(".", "/") + ".class");
+
+        Volume volume = new Volume("/dummy");
+        try {
+            return runInsideDocker(
+                createContainerCmd -> createContainerCmd.withBinds(new Bind(mountableFile.getResolvedPath(), volume, AccessMode.ro)),
+                (__, containerId) -> {
+                    try (InputStream stream = dockerClient.copyArchiveFromContainerCmd(containerId, volume.getPath()).exec()) {
+                        stream.read();
+                        return true;
+                    } catch (Exception e) {
+                        return false;
+                    }
+                }
+            );
         } catch (Exception e) {
-            VisibleAssertions.fail("File should be mountable but fails with " + e.getMessage());
+            log.debug("Failed to check mountable file", e);
```

Comment:
Probably better to rephrase the message: `Failure while checking for mountable file support`

Reply:
So you want to keep the message? :laughing:

Reply:
missed this one :)